skbio/stats/ordination/_principal_coordinate_analysis.py

```diff
@@ -130,20 +162,174 @@ def pcoa(distance_matrix):
 
 
+def _fsvd(centered_distance_matrix, dimension=10,
+          use_power_method=False, num_levels=1):
+    """
+           Performs singular value decomposition, or more specifically in
+           this case eigendecomposition, using fast heuristic algorithm
+           nicknamed "FSVD" (FastSVD), adapted and optimized from the algorithm
+           described by Halko et al (2011).
+
+           Parameters
+           ----------
+           centered_distance_matrix : np.array
+               Numpy matrix representing the distance matrix for which the
+               eigenvectors and eigenvalues shall be computed
+           dimension : int
+               Number of dimensions to keep. Must be lower than or equal to the
+               rank of the given distance_matrix.
+           num_levels : int
+               Number of levels of the Krylov method to use (see paper).
+               For most applications, num_levels=1 or num_levels=2 is
+               sufficient.
+           use_power_method : bool
+               Changes the power of the spectral norm, thus minimizing
+               the error). See paper p11/eq8.1 DOI = {10.1137/100804139}
+
+           Returns
+           -------
+           np.array
+               Array of eigenvectors, each with num_dimensions_out length.
+           np.array
+               Array of eigenvalues, a total number of num_dimensions_out.
+
+           Notes
+           -----
+
+           The algorithm is based on 'An Algorithm for the Principal
+           Component analysis of Large Data Sets'
+           by N. Halko, P.G. Martinsson, Y. Shkolnisky, and M. Tygert.
+           Original Paper: https://arxiv.org/abs/1007.5510
+
+           Ported from reference MATLAB implementation: https://goo.gl/JkcxQ2
+           """
+
+    m, n = centered_distance_matrix.shape
+
+    # Note: a (conjugate) transpose is removed for performance, since we
+    # only expect square matrices.
+    if m != n:
+        raise ValueError('FSVD expects square distance matrix')
+
+    k = dimension + 2
+
+    # Form a real nxl matrix G whose entries are independent,
+    # identically distributed
+    # Gaussian random variables of
+    # zero mean and unit variance
+    G = standard_normal(size=(n, k))
+
+    if use_power_method:
+        # use only the given exponent
+        H = dot(centered_distance_matrix, G)
+
+        for x in range(2, num_levels + 2):
+            # enhance decay of singular values
+            # note: distance_matrix is no longer transposed, saves work
+            # since we're expecting symmetric, square matrices anyway
+            # (Daniel McDonald's changes)
+            H = dot(centered_distance_matrix, dot(centered_distance_matrix, H))
+
+    else:
+        # compute the m x l matrices H^{(0)}, ..., H^{(i)}
+        # Note that this is done implicitly in each iteration below.
+        H = dot(centered_distance_matrix, G)
+        # Again, removed transpose: dot(distance_matrix.transpose(), H)
```

Comment:
Would it be possible to remove the comments about removing code? see also lines 291 295, and maybe others

Reply:
All references to removed code in comments deleted, thanks. I only kept the below comment, because it emphasizes that we expect square matrices.

```
    # Note: a (conjugate) transpose is removed for performance, since we
    # only expect square matrices.
    if m != n:
        raise ValueError('FSVD expects square distance matrix')
```